app/components/checkout/braintree/BraintreePayments.tsx

```diff
@@ -0,0 +1,108 @@
+    const submit = useSubmit();
+
+    const submitPayment = async () => {
+        setProcessing(true);
+        if (braintreeInstance) {
+            try{
+                const result = await braintreeInstance.requestPaymentMethod();
+
+                const formData = new FormData();
+                formData.set("paymentMethodCode", "braintree");
+                formData.set("paymentNonce", result.nonce);
+    
+                let request: Request;
+                request = new Request("");
+    
+                await addPaymentToOrder({ method: "braintree", metadata: result }, { request });
+    
+                submit(formData, { method: "post" });
+            }catch(e){
+                alert(e);
+                setProcessing(false);
+            } 
+        }
+    };
+
+    useEffect(() => {
+        if (show) {
+            const initializeBraintree = () => dropin.create({
+                authorization: authorization,
+                container: '#braintree-drop-in-div',
+                paypal: {
+                    flow: 'checkout',
+                    amount: fullAmount,
+                    currency: currencyCode.toString(),
+                }
+            }, function (error, instance) {
+                if (error)
+                    console.error(error)
+                else if (instance != null) {
+                    setBraintreeInstance(instance);
+                    instance.on("paymentMethodRequestable", (payload) => {
+                        setEnablePaymentButton(true);
+                    });
+                    instance.on("noPaymentMethodRequestable", () => {
+                        setEnablePaymentButton(false);
+                    });
+                }
+            });
+
+            if (braintreeInstance) {
+                braintreeInstance
+                    .teardown()
+                    .then(() => {
+                        initializeBraintree();
+                    });
+            } else {
+                initializeBraintree();
+            }
+        }
+    }, [show])
+
+    return (
+        <div
+            style={{ display: `${show ? "block" : "none"}` }}
+            className={"w-full h-full"}
+        >
+            <div
+                id={"braintree-drop-in-div"}
+            />
+
+            <input
+                type="hidden"
+                name="paymentMethodCode"
+                value={"braintree"}
+            />
+            <button onClick={submitPayment}
+                className={classNames(
+                    enablePaymentButton && !processing
+                        ? 'bg-primary-600 hover:bg-primary-700'
+                        : 'bg-gray-400',
+                    'flex w-full items-center justify-center space-x-2 mt-24 py-3 border border-transparent text-base font-medium rounded-md shadow-sm text-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500',
+                )}disabled={!braintreeInstance || !enablePaymentButton}>
```

Comment:
My IDE complains that there should be a space before `disabled=`, not sure how important that is but I guess it makes sense to add one.